woocommerce/woocommerce
# Update extend-cart-checkout-block dependency

### All Submissions:


### Changes proposed in this Pull Request:

**The proposed changes make sure that the newsletter subscription block will be displayed as expected, as well as enabling developers to use the same formatting WooCommerce Blocks are using and to easily spin up a local WordPress environment.**

While using `npx @wordpress/create-block -t ./woocommerce-monorepo/packages/js/extend-cart-checkout-block/ some-extension-name`, I noticed that the newsletter subscription block, which is part of the examples, didn't appear on the Checkout page. I looked up this issue and noticed that the newsletter subscription block did not appear due to using a wrong handle, which got overwritten by the main package. I also noticed that the dependency was using `"category": "{{slug}}",` instead of `"category": "woocommerce",`, leading to the fact that the newsletter subscription block did not appear as an available block. In addition, I noticed that an SCSS dependency wasn't imported, leading to the fact that one of the examples didn't use the expected styles.

Besides these changes, I've also added the following files:
- `.editorconfig.mustache`: to ensure that the correct editor formatting is used
- `.nvmrc.mustache`: to ensure that the correct Node version is used
- `.prettierrc.js.mustache`: to ensure that the correct Prettier formatting is used, if the optional `wp-prettier` dependency will be installed
- `.wp-env.json.mustache`: to ensure that a local WordPress environment can be created, if the optional `@wordpress/env` dependency will be installed

Finally, I've added steps to the `README.md` for installing the optional `wp-prettier` and `@wordpress/env` dependencies.

### How to test the changes in this Pull Request:


1. Check out this PR.
2. Run the following command in the folder where you want the created cart and checkout extension to appear.
3. Run the following command to create a cart and checkout blocks extension named `some-extension-name`:
```sh
npx @wordpress/create-block -t ./woocommerce-monorepo/packages/js/extend-cart-checkout-block/ some-extension-name
```
(Note: You might have to change the path to where your local WooCommerce repo is located.)
4. Run the following command to format the JS files of your created extension using `wp-prettier`:
```sh
nvm use && npm i --D "prettier@npm:wp-prettier@latest"
```
5. Run the following command to create a local WordPress environment using `wp-env`:
```sh
nvm use && npm i -D @wordpress/env && npm set-script wp-env "wp-env"
```
6. Start Docker Desktop.
7. In the terminal, go to the folder in which the `some-extension-name` extension has been created.
8. Run the following command to start the local WordPress environment:
```sh
npm run wp-env start
```
(Note: If there's already a local WordPress environment running on the ports 8888 and 8889, you'll see an error. In this case, open the Docker Desktop dashboard, delete the existing container and run the previous command again.)
9. Go to http://localhost:8888/wp-admin and log in with the following credentials:
```text
user: admin
pass: password
```
10. Go to http://localhost:8888/wp-admin/edit.php?post_type=page and open the Checkout page.
11. Replace the shortcode with the Checkout block.
12. Verify that you see `I want to receive updates about products and promotions.` below the `Contact Information` inner block.
13. Verify that you see `Data passed to this component: This is some example data from the server` below the total amount and that the font size is bigger and light grey.
14. Create a test product.
15. Go to the frontend, add the test product to the cart and open the Checkout page.
16. Verify that you see the results of 12. and 13. on the Checkout page in the frontend.

### Other information:


### FOR PR REVIEWER ONLY:
